### PR TITLE
Proposal: `Form`-level `onChange` implementation

### PR DESCRIPTION
**Pull request by egeste**
This code is very simple, and it should be easy for anyone to understand. The unit testing framework in place, however, is daunting. If this proposal is acceptable, I would appreciate help getting coverage for this behavior. It is based off of https://github.com/erikras/redux-form/issues/883#issuecomment-215732308

### CODE REVIEW

**Comment by egeste** [comment]
In practice, it seems like `onComponentWillReceiveProps` only gets called for certain fields. For example, in my application, I have several `Field` wrapped components that invoke `this.props.input.onChange` directly with their `value`. For simple component wrappers like `WrappedInput` it works fine, but for more complex ones like `WrappedFileUpload` or `WrappedSelect` (which I'm sorry, I can't share the implementation for yet), `componentWillReceiveProps` does not get invoked. It could be an implementation problem on my end, but I don't think so.

**Comment by egeste** [comment]
ping @erikras

File: src/Form.js [hunk 1]
```diff
@@ -12,6 +13,17 @@ class Form extends Component {
     this.context._reduxForm.registerInnerOnSubmit(this.props.onSubmit)
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.onChange) {
+      const currentValues = this.context._reduxForm.getValues()
+      if (this.previousValues) {
+        const currentValuesDiffer = !deepEqual(currentValues, this.previousValues)
+        if (currentValuesDiffer) nextProps.onChange(currentValues)
```
Comment (gustavohenke) [inline src/Form.js]: I would call `onChange` even for the first call (when `this.previousValues` is falsy)

**Comment by egeste** [comment]
@gustavohenke if you can hack the unit tests in for this, I would be grateful.

**Comment by egeste** [comment]
Another thought - it's probably advisable to always cache `this.previousState`, as adding an `onChange` prop during some point in the component lifecycle will dispatch an errant `onChange` invocation from `undefined` to `currentValues`

**Comment by erikras** [comment]
This is not a bad idea, but you've implemented it in the wrong place.

I've implemented it in #2576.

**Comment by erikras** [comment]
Fix published in [`v6.6.0`](https://github.com/erikras/redux-form/releases/tag/v6.6.0).

**Pull request closed without merging**